src/scores/probability/fss_impl.py

```diff
@@ -0,0 +1,528 @@
+        reduce_dims: Optional set of additional dimensions to aggregate over
+            (mutually exclusive to `reduce_dims`).
+        preserve_dims: Optional set of dimensions to keep (all other dimensions
+            are reduced); By default all dimensions except `spatial_dims`
+            are preserved (mutually exclusive to `preserve_dims`).
+        compute_method: currently only supports `FssComputeMethod.NUMPY`
+            see: :py:class:`FssComputeMethod`
+        dask: See xarray.apply_ufunc for options
+
+    > **CAUTION:** `dask` option is not fully tested
+
+    Returns:
+        An `xarray.DataArray` containing the FSS computed over the `spatial_dims`
+
+        The resultant array will have the score grouped against the remaining
+        dimensions, unless `reduce_dims`/`preserve_dims` are specified; in which
+        case, they will be aggregated over the specified dimensions accordingly.
+
+        For an exact usage please refer to `FSS.ipynb` in the tutorials.
+
+    Raises:
+        DimensionError: Various errors are thrown if the input dimensions
+            do not conform. e.g. if the window size is larger than the input
+            arrays, or if the the spatial dimensions in the args are missing in
+            the input arrays.
+    """
+
+    def _spatial_dims_exist(_dims):
+        s_spatial_dims = set(spatial_dims)
+        s_dims = set(_dims)
+        return s_spatial_dims.intersection(s_dims) == s_spatial_dims and len(spatial_dims) == 2
+
+    if not _spatial_dims_exist(fcst.dims):
+        raise DimensionError(f"missing spatial dims {spatial_dims} in fcst")
+    if not _spatial_dims_exist(obs.dims):
+        raise DimensionError(f"missing spatial dims {spatial_dims} in obs")
+    for s_dim in spatial_dims:
+        if obs[s_dim].shape != fcst[s_dim].shape:
+            raise DimensionError(f"spatial dims {spatial_dims} shapes do not match for fcst and obs")
+
+    # wrapper defined for convenience, since it's too long for a lambda.
+    def fss_wrapper(da_fcst: xr.DataArray, da_obs: xr.DataArray) -> FssDecomposed:
+        fss_backend = FssBackend.get_compute_backend(compute_method)
+        fb_obj = fss_backend(da_fcst, da_obs, threshold=threshold, window=window, zero_padding=zero_padding)
+        return fb_obj.compute_fss_decomposed()
+
+    # apply ufunc to get the decomposed fractions skill score aggregated over
+    # the 2D spatial dims.
+    da_fss = xr.apply_ufunc(
+        fss_wrapper,
+        fcst,
+        obs,
+        input_core_dims=[list(spatial_dims), list(spatial_dims)],
+        vectorize=True,
+        dask=dask,  # pragma: no cover
+    )
+
+    # gather additional dimensions to aggregate over.
+    dims = utils.gather_dimensions2(
+        fcst,
+        obs,
+        weights=None,  # weighted score currently not supported
+        reduce_dims=reduce_dims,
+        preserve_dims=preserve_dims,
+    )
+    dims_reduce = set(dims) - set(spatial_dims)
```

Comment:
“- set(spatial_dims)” handles that

Reply:
As discussed will add a `warning` here. Because user may be intending to (or accidentally) preserve `spatial_dims` (or subset of) when that's not actually possible for this particular score.